core/jobs/batch_jobs/voiceover_migration_job.py

```diff
@@ -0,0 +1,358 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs used for migrating voiceovers from exploration models to entity
+voiceover models."""
+
+from __future__ import annotations
+
+from core.domain import state_domain
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+from typing import Dict, List, Tuple, Union
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import datastore_services
+    from mypy_imports import exp_models
+    from mypy_imports import voiceover_models
+
+datastore_services = models.Registry.import_datastore_services()
+
+(voiceover_models, exp_models) = models.Registry.import_models([
+    models.Names.VOICEOVER, models.Names.EXPLORATION])
+
+ExplorationAndVoiceArtistLinkType = Tuple[str, Dict[str, List[Union[
+    voiceover_models.ExplorationVoiceArtistsLinkModel,
+    exp_models.ExplorationModel]]]]
+
+
+class PopulateManualVoiceoversToEntityVoiceoverModelJob(base_jobs.JobBase):
+    """Migrates voiceovers stored in the exploration model to the new
+    EntityVoiceovers storage models.
+    """
+
+    DATASTORE_UPDATES_ALLOWED = True
+
+    @classmethod
+    def create_entity_voiceovers_model_instance(
+        cls,
+        entity_type: str,
+        entity_id: str,
+        entity_version: int,
+        language_accent_code: str
+    ) -> voiceover_models.EntityVoiceoversModel:
+        """Returns a unique entity voiceovers model instance.
+
+        Args:
+            entity_type: str. The type of the entity.
+            entity_id: str. The ID of the entity.
+            entity_version: int. The version of the entity.
+            language_accent_code: str. The language-accent code of the
+                voiceover.
+
+        Returns:
+            EntityVoiceoversModel. An instance of the entity voiceover model.
+        """
+        with datastore_services.get_ndb_context():
+            entity_voiceovers_model = (
+                voiceover_models.EntityVoiceoversModel.create_new(
+                    entity_id, entity_type, entity_version,
+                    language_accent_code, {}
+                )
+            )
+        entity_voiceovers_model.update_timestamps()
+        return entity_voiceovers_model
+
+    @classmethod
+    def update_entity_voiceover_for_given_id(
+        cls,
+        entity_type: str,
+        entity_id: str,
+        entity_version: int,
+        accent_code: str,
+        content_id: str,
+        voiceover_dict: state_domain.VoiceoverDict,
+        entity_voiceover_id_to_entity_voiceovers: Dict[
+            str, voiceover_models.EntityVoiceoversModel]
+    ) -> Dict[str, voiceover_models.EntityVoiceoversModel]:
+        """Updates the entity voiceover model instance for the given language
```

Comment:
I can't tell what this method is supposed to be doing. What sort of update is being made -- can you be more specific?